pages/index.tsx

```diff
@@ -384,10 +394,13 @@ export default function Home({ featuredStorefronts }: HomeProps) {
                   <SelectInline
                     value={filterBy}
                     label="Filter"
-                    onChange={(next) => {
-                      const filter = next as FilterOptions;
+                    onChange={(nextFilter) => {
+                      const filter = nextFilter as FilterOptions;
                       setFilterBy(filter);
-                      setSortBy(sortOptions[filter][0].key);
+                      // only reset sortBy if it does not work in the new filter
+                      if (!sortOptions[filter].find((s) => s.key === sortBy)) {
```

Comment:
👍 

```
pipe(any(pipe(prop('key'), equals(sortBy)), not)(sortOptions[filter])
```

If you want to compose with ramda but I can see this being less readable so no biggie leaving out.